components/Marketplace/CreateListingDrawer.tsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import {
+  Drawer,
+  TextInput,
+  Textarea,
+  Button,
+  Group,
+  Select,
+  Text,
+  Title,
+  NumberInput,
+  Grid,
+} from '@mantine/core';
+import { notifications } from '@mantine/notifications';
+import { useDisclosure } from '@mantine/hooks';
+import { useFormik } from 'formik';
+import { Visibility } from '@/src/API';
+import { useCreateListing } from '@/src/hooks/marketplaceCustomHooks';
+
+interface CreateListingDrawerProps {
+  opened: boolean;
+  onClose: () => void;
+  onPostCreated: () => void;
+}
+
+export function CreateListingDrawer({ opened, onClose, onPostCreated }: CreateListingDrawerProps) {
+  const [loading, handlers] = useDisclosure();
+  const { handleCreateListing } = useCreateListing();
+
+  const formik = useFormik({
+    initialValues: {
+      title: '',
+      price: '',
+      description: '',
+      contact: '',
+      visibility: Visibility.PUBLIC,
+    },
+    // validationSchema: createEventSchema,
```

Comment:
Did we need a createListingSchema here as well?

Reply:
Yup, changed in Demessie's ticket tho